Sources/Shared/Classes/DataSource.swift

```
@@ -0,0 +1,5 @@
+import Foundation
+
+public class DataSource: NSObject {
+  weak var spot: Spotable!
```

@vadymmarkov Then we need to optional assign `spot` in all data source and delegate methods but maybe that's worth it? 😁 Which would mean that it would "silent fail" instead of blow up into a million pieces 💥